### DISCUSSION TITLE
How to pass array attributes to Markdoc tag?

### DISCUSSION THREAD

**Discussion by naderkhalil** [markdoc/markdoc#258] [Q&A] [answered]
The [docs](https://markdoc.dev/docs/attributes) mention setting the attribute type as array:

```
TabbedContent: {
    attributes: {
      tabs: {
        type: Array,
        required: true,
        errorLevel: 'critical',
        default: [],
      }
    }
  },
```
  
  Then in markdown:
  
```
{% TabbedContent tabs={["hello"]} %}
{% /TabbedContent %}
```

But I get the following error:
```
Error: Expected "}", identifier, string, or whitespace but "[" found.
{% TabbedContent tabs={["hello"]} %}
                       ^

Node 'tag' is missing opening
{% /TabbedContent %}
```

Am I doing something wrong? I couldn't find an example of Array used in the docs. Thank you!!

**Comment by varughese** [comment, marked as answer]
Hey! This should work:

{% TabbedContent tabs=["hello"] %}

**Reply by naderkhalil** [reply]
Hey @varughese! Thank you for the speedy reply :) 

Yes! removing the curly braces worked. Is it possible to pass in an array of objects? 

```
{% TabbedContent tabs=[{...}, {...}] %}
```

**Reply by varughese** [reply]
Yup that should work!

If you use the sandbox, you can look at the AST tab to see how Markdoc is parsing the content:

https://markdoc.dev/sandbox?c=LS0tCnRpdGxlOiBXaGF0IGlzIE1hcmtkb2M%252FCi0tLQoKeyUgdGFiYmVkLWNvbnRlbnQgYXJyPVt7Zm9vOiAiYmFyIn0sIHtmb286ICJiYXoifV0gLyV9&mode=ast

**Reply by varughese** [reply]
For what is worth, for tabs I think a more ergonomic solution for tabs would be something like


```
{% tabs %}
{% tab title="A" %}
Content A
{% /tab %}
{% tab title="B" %}
Content B
{% /tab %}
{% /tabs %}
```

**Reply by naderkhalil** [reply]
I totally agree that's a much better way. In case someone else lands on this discussion, I think what was causing the array of objects to fail for me was single quotes:

`{% Component arr=[{title: 'Title'}] %}` should be `{% Component arr=[{title: "Title"}] %}`

**Reply by naderkhalil** [reply]
For the ergonomic solution you proposed, how would "tabs" component manage the state of the "tab" that's selected?

**Reply by varughese** [reply]
Check out https://markdoc.dev/docs/examples#tabs !

We should also make that error message clearer!

**Reply by naderkhalil** [reply]
Thank you!!!
